sdks/python/src/opik/integrations/harbor/opik_tracker.py

```diff
@@ -0,0 +1,473 @@
+"""
+Opik tracking integration for Harbor benchmark evaluation framework.
+
+This module provides the `track_harbor` function to add Opik tracing to Harbor Jobs,
+enabling real-time streaming of trial results to Opik for visualization and analysis.
+
+Example:
+    >>> from opik.integrations.harbor import track_harbor
+    >>> from harbor.job import Job
+    >>> import os
+    >>>
+    >>> os.environ["OPIK_PROJECT_NAME"] = "swebench-evaluation"
+    >>>
+    >>> job = Job(config)
+    >>> tracked_job = track_harbor(job)
+    >>> result = await tracked_job.run()
+
+Or enable tracking globally (for CLI usage):
+    >>> from opik.integrations.harbor import track_harbor
+    >>> track_harbor()
+    >>> # Now run Harbor code - it will be traced
+"""
+
+import functools
+import logging
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, TYPE_CHECKING
+
+from opik import context_storage, id_helpers, opik_context, track
+from opik.api_objects import opik_client
+from opik.types import FeedbackScoreDict, SpanType
+
+from . import experiment_service
+
+if TYPE_CHECKING:
+    from harbor.job import Job
+    from harbor.trial.trial import Trial
+    from harbor.models.trial.result import TrialResult
+    from harbor.models.trajectories.step import Step
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _rewards_to_feedback_scores(
+    rewards: Optional[Dict[str, Any]],
+    error: Optional[str] = None,
+) -> List[FeedbackScoreDict]:
+    """Convert Harbor verifier rewards to Opik feedback scores."""
+    if rewards is None:
+        return []
+
+    feedback_scores: List[FeedbackScoreDict] = []
+    for name, value in rewards.items():
+        try:
+            float_value = float(value)
+
+            score = FeedbackScoreDict(name=name, value=float_value, reason=error)
+
+            feedback_scores.append(score)
+        except (ValueError, TypeError):
+            LOGGER.warning(
+                "Could not convert reward value to float: %s=%s", name, value
+            )
+
+    return feedback_scores
+
+
+def _parse_iso_timestamp(timestamp_str: Optional[str]) -> Optional[datetime]:
```

Comment:
Potentially useful utility function for other integrations too, let's move it to `opik.datetime_helpers`